ADRs/ADR0006-replace-badger-kv-store [WITHDRAWN].md

```diff
@@ -0,0 +1,225 @@
+* Good, because it guarantees atomic transactions.
+* Good, because it easily handles more complex values.
+* Bad, because we want strict hash comparisons to continue working and JSONB interpretation could have unforeseen consequences.
+* Bad, because it might not have the same performance capabilities as Badger.
+* Bad, because it's Postgres specific which could impose limitations on us in the future or be a source of *unknown unknowns*.
+
+### [Option 1.3] - Store Keys In KV Table
+
+Option 3 would dump everything into typical TEXT types in a standard table named `transaction`.
+
+* Good, because it's simple and straight-forward table access. JSONB and HSTORE have new, Postgres specific syntax and behavior.
+* Good, because it stays within standard SQL features, and doesn't utilize Postgres specific behavior.
+* Good, because it handles prefix queries.
+* Bad, because it lacks similar performance capabilities compared to Badger.
+
+### [Option 2.1] - Make KV Store Handle Atomic Guarantees
+
+Option 2 would involve extending the `KVStore` interface so that it can be atomically controlled by the caller (i.e. TxIndexer, block store, or the state trees) either by having it directly extend the Rollback & Savepoint interface, as demonstrated below, or by having it manage its own rollbacks.
+
+```go
+func (indexer *txIndexer) Rollback([]byte) error {
+  return fmt.Errorf("not impl")
+}
+func (indexer *txIndexer) Savepoint() ([]byte, error) {
+  return nil, fmt.Errorf("not impl")
+}
+```
+
+This option is similar to [PR #657 - KV POC](https://github.com/pokt-network/pocket/pull/657) but different in a key way: it addresses the core design misalignment between the `PersistenceRWContext` and the `KVStore` causing the dissonance in the first place.
+
+* Good, because it avoids migrations.
+* Good, because it clears up KVStore's responsiblity boundary as a component in this system.
+* Good, because it takes advantage of existing Badger features.
+* Bad, because it could be more complex than just passing it off to Postgres.
+
+### [Option 3] - Manually Cleanup Leftover Keys
+
+This option is to manually remove or delete keys from Badger that are added in the middle of a rollback.
+
+* Bad, because manually handling cases like this is error prone and this issue directly effects chain integrity.
+* Good, because it requires smaller and fewer changes to implement.
+
+### [Option 4] - Do Nothing
+
+One should always consider the option of doing nothing. In this case, doing nothing results in orphaned keys being left in the Badger KV store.
+
+* Good, because it is always easy to continue in the same pattern.
+* Good, because it fully leverages the performance of Badger.
+* Good, because it avoids a new table in the Postgres database, which avoids a migration.
+* Bad, because it doesn't allow us to easily synchronize an atomic backup or sync and leads to continued orphaned keys in the database.
+* Bad, because it currently leaves orphaned keys in the database.
```

Comment:
Also, `Bad, because it could cause issues in block validation and normal downstream node operations after an error occured`